src/picom.c

```diff
@@ -991,15 +991,31 @@ void root_damaged(session_t *ps) {
 		if (ps->root_image) {
 			ps->backend_data->ops->release_image(ps->backend_data, ps->root_image);
 		}
+
 		auto pixmap = x_get_root_back_pixmap(ps->c, ps->root, ps->atoms);
 		if (pixmap != XCB_NONE) {
+			xcb_generic_error_t *e;
+			auto r = xcb_get_geometry_reply(
+			    ps->c, xcb_get_geometry(ps->c, pixmap), &e);
+			if (!r) {
+				x_print_error(e->full_sequence, e->major_code,
+				              e->minor_code, e->error_code);
+				free(e);
+				goto err;
+			}
+
 			ps->root_image = ps->backend_data->ops->bind_pixmap(
-			    ps->backend_data, pixmap, x_get_visual_info(ps->c, ps->vis), false);
+			    ps->backend_data, pixmap,
+			    x_get_visual_info(ps->c, x_get_visual_for_depth(ps->c, r->depth)),
```

Comment:
i had to do some research to figure out why we couldn't do the same as `win_bind_pixmap` and use the pixmap's visual directly. 

turns out the visual is part of the window's attributes, and GetWindowAttributes doesn't work with a pixmap. can you add a comment to record this fact? thanks.

Reply:
i've tried to add a descriptive comment what we are doing and why.